mingw.invoke.h

```diff
@@ -21,27 +21,35 @@ namespace mingw_stdthread
 {
 namespace detail
 {
-//  For compatibility, implement std::invoke for C++11 and C++14
-#if __cplusplus < 201703L
   template<bool PMemFunc, bool PMemData>
   struct Invoker
   {
     template<class F, class... Args>
+#if __cplusplus >= 201703L
+    inline static typename std::invoke_result<F, Args...>::type invoke (F&& f, Args&&... args)
+#else
     inline static typename std::result_of<F(Args...)>::type invoke (F&& f, Args&&... args)
+#endif
     {
+#if __cplusplus >= 201703L
+      return std::__invoke(std::forward<F>(f), std::forward<Args>(args)...);
+#else
       return std::forward<F>(f)(std::forward<Args>(args)...);
+#endif
     }
   };
+
   template<bool>
   struct InvokerHelper;
 
   template<>
   struct InvokerHelper<false>
   {
     template<class T1>
-    inline static auto get (T1&& t1) -> decltype(*std::forward<T1>(t1))
+    inline static auto get (T1&& t1) ->\
+     decltype(std::forward<T1>(t1))
     {
-      return *std::forward<T1>(t1);
+      return std::forward<T1>(t1);
```

Comment:
Your change here appears to violate ["If t1 does not satisfy the previous items, then INVOKE(f, t1, t2, ..., tN) is equivalent to ((*t1).*f)(t2, ..., tN)."](https://en.cppreference.com/w/cpp/utility/functional/invoke) from the C++ standard.

Reply:
That's why line 66 is different for true.

Reply:
The change you submitted in the line that I highlighted will break the case of `invoke(pointer_to_data_member, pointer_to_object)`. The change on line 66 does not help, due to being a pure whitespace change.

To borrow and modify a small piece of the example from [cppreference.com](https://en.cppreference.com/w/cpp/utility/functional/invoke), the standard indicates that we can write:
```
struct Foo {
    int num_ = 314159;
};
 
int main()
{ 
    using namespace mingw_stdthread::detail;
    // invoke a member function
    const Foo foo(314159);
 
    // invoke (access) a data member using a pointer to the object.
    std::cout << "num_: " << invoke(&Foo::num_, &foo) << '\n';
}
```

This should compile without issues in the base mingw-std-threads library (with appropriate header inclusions, of course). However, I expect your modification to cause a compilation error.

Reply:
Thanks for the explanations. I already know that std :: result_of is complemented with invoke_result functions. :)